src/main/java/com/breskul/bibernate/persistence/EntityManagerImpl.java

```diff
@@ -65,7 +66,42 @@ public void persist(Object entity) {
     @Override
     public <T> T merge(T entity) {
         validateSession();
-        return null;
+        Object id = getIdentifierValue(entity);
+        Class<T> entityClass = (Class<T>) entity.getClass();
+        T newEntity;
+
+        if (cache.containsKey(EntityKey.of(entityClass, id))) {
+            return entity;
+        }
+
+        newEntity = find(entityClass, id);
+        mergeEntities(newEntity, entity);
+
+        return newEntity;
+    }
+
+    private <T> void mergeEntities(T newEntity, T oldEntity) {
```

Comment:
The logic for merge `mergeEntities ` appears to be correct. However, we never send updates to database. Is this intentional ?